whois-update/src/main/java/net/ripe/db/whois/update/dns/zonemaster/ZonemasterDnsGateway.java

```diff
@@ -0,0 +1,164 @@
+package net.ripe.db.whois.update.dns.zonemaster;
+
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.Uninterruptibles;
+import net.ripe.db.whois.common.Message;
+import net.ripe.db.whois.common.profiles.DeployedProfile;
+import net.ripe.db.whois.update.dns.DnsCheckRequest;
+import net.ripe.db.whois.update.dns.DnsCheckResponse;
+import net.ripe.db.whois.update.dns.DnsGateway;
+import net.ripe.db.whois.update.dns.zonemaster.domain.GetTestResultsRequest;
+import net.ripe.db.whois.update.dns.zonemaster.domain.GetTestResultsResponse;
+import net.ripe.db.whois.update.dns.zonemaster.domain.StartDomainTestRequest;
+import net.ripe.db.whois.update.dns.zonemaster.domain.StartDomainTestResponse;
+import net.ripe.db.whois.update.dns.zonemaster.domain.TestProgressRequest;
+import net.ripe.db.whois.update.dns.zonemaster.domain.TestProgressResponse;
+import net.ripe.db.whois.update.domain.UpdateMessages;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import javax.ws.rs.ProcessingException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static net.ripe.db.whois.common.Messages.Type.ERROR;
+
+@DeployedProfile
+@Component
+public class ZonemasterDnsGateway implements DnsGateway {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ZonemasterDnsGateway.class);
+
+    private static final int TEST_PROGRESS_SLEEP_SECONDS = 1;
+    private static final int TEST_PROGRESS_MAXIMUM_RETRIES = 60 * 5;
```

Comment:
I suggest increasing the sleep time then pinging for a result fewer times. e.g. sleep 5 sec and retry 50 times (250 sec in total)